### ISSUE TITLE
Raindrops: Current test suite doesn't include edge cases

### ISSUE THREAD

**Issue by McEileen** [exercism/xruby#434]
My original solution for Raindrops passed all the Version 2 tests, however, it failed certain edge cases that weren't in the original test suite.  At first I solved the problem using the 'prime' module and prime_division. The tests didn't catch this, but my original solution had errors for numbers that are raised to the 3rd, 5th or 7th power, but don't have those numbers as factors. For example, 2187 (3 to the 7th power) would return "PlingPlong" instead of "Pling."  I'm going to submit a pull request with tests for these edge cases.

**Comment by kotp** [comment]
The simplest test for this may cover using Prime, but also ensures that the order of replacement is good for any size combination of factors.

```ruby
def test_should_give_all_replacements_in_order_for_non_prime_0
  assert_equal 'PlingPlangPlong', Raindrops.convert(0)
end
```

What would that test give in your version?

**Comment by Insti** [comment]
The tests still need updating with the test cases from the common data.

**Comment by kotp** [comment]
@McEileen Are you interested in making this generated?

**Comment by Insti** [comment]
Fixed by #441

**Comment by McEileen** [comment]
Hi, I didn't read the comments until now, sorry.  Thanks for working on it, @Insti!
